File: src/LinksNews/Services/Data/Models/MessageRecipient.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace LinksNews.Services.Data.Models
{
    [Table("messageRecipient")]
    public class MessageRecipient
    {
        public long? Id { get; set; }
        public long? MessageId { get; set; }
        public long? RecipientAccountId { get; set; }
        public string RecipientAddress { get; set; }
        public DateTimeOffset? ReceiveDate { get; set; }
        public long? CommunicationMethodId { get; set; }
    }
}

File: src/LinksNews/angular2App/app/core/domain/message-recipient.ts
import * as _ from 'lodash';

export class MessageRecipient {
    id: number;
    messageId: number;
    recipientAccountId: number;
    recipientAddress: string;
    receiveDate: Date;
    communicationMethodId: number;

    clone(): MessageRecipient {
        let result: MessageRecipient = <MessageRecipient>_.assign(new MessageRecipient(), this);
        return result;
    }

    static from (account: MessageRecipient): MessageRecipient {
        let result: MessageRecipient = new MessageRecipient();
        _.assign(result, account);
        return result;
    }
}
